File: runs/A/bmad/02/code/seed-a-frontend/src/app/core/api/booking.ts
import { PriceCategory } from './event-detail';

export type BookingStatus = 'aktiv' | 'storniert';

export interface BookingPosition {
  rowLabel: string;
  columnNumber: number;
  preiskategorie: PriceCategory;
}

export interface Booking {
  reference: string;
  name: string;
  status: BookingStatus;
  positionen: BookingPosition[];
  gesamtpreis: number;
}

export interface CreateBookingPositionRequest {
  rowLabel: string;
  columnNumber: number;
  priceCategoryId: number;
}

export interface CreateBookingRequest {
  eventId: number;
  name: string;
  email: string;
  positionen: CreateBookingPositionRequest[];
}

export function parseSeatCode(seatCode: string): { rowLabel: string; columnNumber: number } {
  const match = /^([A-Za-z]+)(\d+)$/.exec(seatCode);
  if (!match) {
    throw new Error(`Ungültiger Sitzplatz-Code: '${seatCode}'`);
  }
  return { rowLabel: match[1], columnNumber: Number(match[2]) };
}

File: runs/A/bmad/01/code/seed-a-backend.Api/Domain/Booking.cs
namespace seed_a_backend.Api.Domain;

public enum BookingStatus
{
    Active,
    Cancelled
}

public class Booking
{
    public Guid Id { get; set; }
    public required string Reference { get; set; }
    public Guid EventId { get; set; }
    public Event? Event { get; set; }
    public required string Name { get; set; }
    public required string Email { get; set; }
    public BookingStatus Status { get; set; }
    public ICollection<BookingSeat> Seats { get; set; } = new List<BookingSeat>();
}
